core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/recoverer.go

```diff
@@ -112,34 +114,40 @@ func (r *logRecoverer) Start(pctx context.Context) error {
 	r.lock.Unlock()
 
 	blockTimeResolver := newBlockTimeResolver(r.poller)
-	blockTime, err := blockTimeResolver.BlockTime(ctx, defaultSampleSize)
-	if err != nil {
-		// TODO: TBD exit or just log a warning
-		// return fmt.Errorf("failed to compute block time: %w", err)
-		r.lggr.Warnw("failed to compute block time", "err", err)
-	}
-	if blockTime > 0 {
-		r.blockTime.Store(int64(blockTime))
+	updateBlockTime := func(ctx context.Context) {
+		blockTime, err := blockTimeResolver.BlockTime(ctx, defaultSampleSize)
+		if err != nil {
+			r.lggr.Warnw("failed to compute block time", "err", err)
+			return
+		}
+		if blockTime > 0 {
+			r.blockTime.Store(int64(blockTime))
+		}
 	}
+	updateBlockTime(ctx)
 
 	r.lggr.Infow("starting log recoverer", "blockTime", r.blockTime.Load(), "lookbackBlocks", r.lookbackBlocks.Load(), "interval", r.interval)
 
 	{
 		go func(ctx context.Context, interval time.Duration) {
-			ticker := time.NewTicker(interval)
-			defer ticker.Stop()
+			recoverTicker := time.NewTicker(interval)
+			defer recoverTicker.Stop()
 			gcTicker := time.NewTicker(utils.WithJitter(GCInterval))
 			defer gcTicker.Stop()
+			blockTimeUpdateTicker := time.NewTicker(blockTimeUpdateCadence)
+			defer blockTimeUpdateTicker.Stop()
 
 			for {
 				select {
-				case <-ticker.C:
+				case <-recoverTicker.C:
 					if err := r.recover(ctx); err != nil {
 						r.lggr.Warnw("failed to recover logs", "err", err)
 					}
 				case <-gcTicker.C:
 					r.clean(ctx)
 					gcTicker.Reset(utils.WithJitter(GCInterval))
+				case <-blockTimeUpdateTicker.C:
+					updateBlockTime(ctx)
```

Comment:
We might need to spawn a goroutine for the block time update as it could block (makes DB calls)
Same for `clean()` so we can also do it in a separate PR